PSReadLine/ReadLine.cs

```diff
@@ -144,6 +144,14 @@ private void ReadOneOrMoreKeys()
                 while (_charMap.KeyAvailable)
                 {
                     var key = PSKeyInfo.FromConsoleKeyInfo(_charMap.ReadKey());
+                    // If PSReadLine is running under a host that can cancel it, the cancellation
+                    // will come at a time when ReadKey is stuck waiting for input. The next key
+                    // press will be used to force it to return, and so we want to discard this key
+                    // since we were already canceled.
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
Minor comment: maybe update this code a little bit to the following:
```
ConsoleKeyInfo kyeInfo = _charMap.ReadKey();
if (cancellationToken.IsCancellationRequested)
{
    // If PSReadLine is running under a host that can cancel it, the cancellation
    // will come at a time when ReadKey is stuck waiting for input. The next key
    // press will be used to force it to return, and so we want to discard this key
    // since we were already canceled.
    break;
}

var key = PSKeyInfo.FromConsoleKeyInfo(kyeInfo);
_lastNKeys.Enqueue(key);
_queuedKeys.Enqueue(key);
```

Reply:
👍